lib/project.ts

```diff
@@ -368,22 +373,13 @@ export class Project implements Project.IProject {
 	}
 
 	public getConfigurationsSpecifiedByUser(): string[] {
-		let configurations: string[] = [];
-		if (this.$options.debug) {
-			configurations.push(this.$projectConstants.DEBUG_CONFIGURATION_NAME);
-		}
-
-		if (this.$options.release) {
-			configurations.push(this.$projectConstants.RELEASE_CONFIGURATION_NAME);
+		return this.$options.config || [];
```

Comment:
If the user enters invalid configuration we should display warning which says that the invalid configuration does not exist and the changes for it will not be applied or throw an error with message for invalid configuration. If we display warning we should remove the invalid configurations and when this method is used it will return only valid configurations.